src/provider.c

```diff
@@ -1139,7 +1147,7 @@ int OSSL_provider_init(const OSSL_CORE_HANDLE *handle, const OSSL_DISPATCH *in,
 
     /* get module path */
     core_params[0] = OSSL_PARAM_construct_utf8_ptr(
-        P11PROV_PKCS11_MODULE_PATH, (char **)&ctx->module, sizeof(ctx->module));
+        P11PROV_PKCS11_MODULE_PATH, &ctx->module, sizeof(ctx->module));
```

Comment:
This will not copy the path I think, only reference the internal storage, so you need to copy it after the fact I think, otherwise the free() at line 1058 may break things

Reply:
You are probably right. This API is quite opaque so it is hard to guess what does it actually do, but this would make sense.